### PR TITLE
Use new VS Code test package

### PR DESCRIPTION
**Pull request by alexweininger**

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -16,10 +16,10 @@ let perfStats = {
 
 Object.defineProperty(exports, "__esModule", { value: true });
 
-const extension = require('./out/src/extension');
+const extension = require('./dist/extension.bundle');
```
Comment (nturinski) [inline main.js:19]: I don't think you meant to check this in?

Comment (alexweininger) [reply main.js:19]: Need to undo it here too. Usually I catch this... https://github.com/microsoft/vscode-azurefunctions/pull/3275/files#diff-58417e0f781b6656949d37258c8b9052ed266e2eb7a5163cad7b0863e6b2916a

File: package.json [hunk 2]
```diff
@@ -1,812 +1,819 @@
 {
-    "name": "vscode-azureappservice",
```
Comment (bwateratmsft) [inline package.json:2]: Your formatter replaced all the four-space indents with tabs; you'll want to revert that.

File: test/runTest.ts [hunk 3]
```diff
@@ -3,21 +3,39 @@
...
  *--------------------------------------------------------------------------------------------*/
 
+import { downloadAndUnzipVSCode, resolveCliArgsFromVSCodeExecutablePath, runTests } from '@vscode/test-electron';
+import * as cp from 'child_process';
 import * as path from 'path';
-import { runTests } from 'vscode-test';
 
 async function main(): Promise<void> {
     try {
+        const vscodeExecutablePath = await downloadAndUnzipVSCode('1.69.0');
```
Comment (bwateratmsft) [inline test/runTest.ts:12]: Why hardcode a version here?

File: package.json [hunk 4]
```diff
@@ -808,5 +809,11 @@
     ],
     "enabledApiProposals": [
         "terminalDataWriteEvent"
-    ]
+    ],
+    "__metadata": {
```
Comment (bwateratmsft) [inline package.json:813]: What's this for?

File: package.json [hunk 5]
```diff
@@ -795,6 +795,7 @@
         "@microsoft/vscode-azext-azureappservice": "^0.7.0",
         "@microsoft/vscode-azext-azureutils": "^0.3.4",
         "@microsoft/vscode-azext-utils": "^0.3.7",
+        "@vscode/test-electron": "^2.1.5",
```
Comment (bwateratmsft) [inline package.json:798]: This should go into the `devDependencies` section, replacing `vscode-test`.

File: package.json (hunk 4 above)
Comment (alexweininger) [reply package.json:813]: When I run `npm run test` it adds this. It also changes the spaces to tabs.

Comment (alexweininger) [reply package.json:812]: Yeah, it's caused by the @vscode/test-electron package. Other extensions just don't commit the changes. Sorta annoying...

https://github.com/microsoft/vscode/issues/148975

Comment (bwateratmsft) [reply package.json:812]: Ugh. Well, try removing that and see if the tests run successfully.

**Review by bwateratmsft — APPROVED** [review]

**Pull request merged**